src/_pytest/assertion/rewrite.py

```diff
@@ -128,12 +131,24 @@ def create_module(
         return None  # default behaviour is fine
 
     def exec_module(self, module: types.ModuleType) -> None:
-        assert module.__spec__ is not None
-        assert module.__spec__.origin is not None
-        fn = Path(module.__spec__.origin)
+        co = self.get_code(module.__name__)
+        exec(co, module.__dict__)
+
+    def get_code(self, fullname: str) -> Optional[types.CodeType]:
+        """
+        Returns the code for a rewritten test file.
+        """
+        # The type annotation is *Optional[...]* to match the declaration
+        # in importlib.abc.InspectLoader
+
+        try:
+            fn = Path(self._fullname_to_info[fullname][0])
+        except KeyError:
+            raise ImportError(fullname)
```

Comment:
is this error possible? this shouldn't be needed in a PEP451 loader (the code retrieval step doesn't happen if the finder isn't returned) -- this makes me think that we've reverted back to 302

Reply:
yeah InspectLoader is the 302 interface which would regress us back into `imp` land